File: Dice/Dice.py
# Author @TateSamuD
# Date 08/11/2023
import random


def main():
    # TODO add roll loop
    chc = int(input("How many sides do you want your dice to have: "))

    # TODO fix input verification
    # if chc != int:
    #     while chc != int:
    #         print("Please enter a number.")
    #         chc = int(input("How many side do you want the dice to have: "))

    outcome = roll(chc)

    print("You rolled a %i" % outcome)

    # input how many sides the dice needs to have

    # use input as the upper bound for the random generator


def roll(uBound):
    throw = random.randint(0, uBound)
    return throw


if __name__ == "__main__":
    main()

File: Dice/Dice.java
package Dice;

/*
 * @author Tatenda Samudzi
 */
import java.util.*;

public class Dice {
    int Roll(int num) {
        int val = 0;
        Random rand = new Random();
        val = rand.nextInt(num);
        return val;
    }

    public static void main(String[] args) {
        Scanner scnr = new Scanner(System.in);
        Dice dice = new Dice();
        System.out.println("Enter the number of sides you want the dice to have: ");
        int uBound = scnr.nextInt();
        int outCome = dice.Roll(uBound);
        scnr.close();
        System.out.printf("You rolled a %d\n", outCome);
        System.out.printf("Thank you!");
    }
}